tests/test_virtualenv.py

```diff
@@ -35,6 +36,47 @@ def factory(*args, **kwargs):
     return factory
 
 
+@pytest.fixture
+def make_mocked_interpreter_path():
+    def factory(path, sysexec_result):
+        def mock_sysexec(*_):
+            if sysexec_result == RAISE_ERROR:
+                raise py.process.cmdexec.Error(1, 1, "", "", "")
+            else:
+                return sysexec_result
+
+        attrs = {
+            "sysexec.side_effect": mock_sysexec,
+            "__str__": mock.Mock(return_value=path),
+        }
+        mock_python = mock.Mock()
+        mock_python.configure_mock(**attrs)
```

Comment:
Ditto here, is there some reason we can't use `__spec__`?

Reply:
I understood that a plain `Mock()` would be preferable if possible (and `MagicMock(spec=("__str__"))` if it's not possible). I admit I'm unsure what exactly you are referring to by "Ditto" and "__spec__". I can change it to something like `MagicMock(spec=("__str__"))`? I might be confusing things here, also due to my lack of experience with `unittest.mock`, if in doubt, could you suggest a code example? Thanks!